### PR TITLE
Add regexp operator tests

### PR DESCRIPTION
**Pull request by superkhau**
Add tests to ensure regexp operator works properly when flags are provided.

### CODE REVIEW

**Comment by superkhau** [comment]
@raymondfeng Merging without review, hardening with more tests.

File: test/postgresql.test.js [hunk 1]
```diff
@@ -206,6 +206,15 @@ describe('postgresql connector', function () {
           console.warn.restore();
         });
 
+        it('should work', function(done) {
+          Post.find({where: {content: {regexp: '^a/i'}}}, function(err, posts) {
```
Comment (raymondfeng) [inline test/postgresql.test.js]: Is the leading `/` optional? JS uses `/^a/i` as the literal.

Comment (superkhau) [inline test/postgresql.test.js]: For regex strings it is. My toRegExp normalization will convert it to a RegExp object properly (which in turn converts '^a/i' to /a/i underneath the hood.

Comment (superkhau) [inline test/postgresql.test.js]: Also its not really just the leading slash, its more "you don't need to provide regex delimiters for regex strings" (meaning the leading and trailing forward slashes).

Comment (raymondfeng) [inline test/postgresql.test.js]: ok, I wonder if we want to be strict, i.e., the string should be:

1. /pattern/
2. /pattern/i
3. pattern

I'm fine to allow `pattern/i` if it doesn't cause confusions.

Comment (superkhau) [inline test/postgresql.test.js]: Hm...good question. `pattern/i` is actually the syntax MySQL uses. I believe I coded defensively, so maybe just don't announce it in the docs, but let it slide?

**Pull request merged**